windows/src/global/delphi/comp/KeymanDeveloperDebuggerMemo.pas

```diff
@@ -28,11 +28,18 @@ interface
 type
   TKeymanDeveloperDebuggerMessageEvent = procedure(Sender: TObject; var Message: TMessage; var Handled: Boolean) of object;
 
+  TMemoSelection = record
```

Comment:
All changes in this file provide values for:

(1) Correct handling of text selection and deletion
(3) Deadkey realignment after text changes
(4) correct tracking of cursor position between events